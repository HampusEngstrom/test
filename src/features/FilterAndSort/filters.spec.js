import {
  filterOnSingleChoice,
  filterOnTextInput,
  filterOnMultipleChoices,
} from './filters';

describe('filter on single choice', () => {
  const items = [
    { id: '1', a: 'hej', b: 'hej' },
    { id: '2', a: 'hej', b: null },
    { id: '3', a: null, b: 'hej' },
    { id: '4', a: 'hej', b: 'hej' },
    { id: '5', a: 'hej', b: null },
    { id: '6', a: null, b: 'hej' },
  ];

  it('filterOnSingleChoice returns correct a b', () => {
    const map = {
      a: true,
      b: true,
    };
    const result = items.filter(filterOnSingleChoice(map));
    expect(result).toMatchSnapshot();
  });

  it('filterOnSingleChoice returns correct a', () => {
    const map = {
      a: true,
      b: false,
    };
    const result = items.filter(filterOnSingleChoice(map));
    expect(result).toMatchSnapshot();
  });

  it('filterOnSingleChoice returns correct b', () => {
    const map = {
      a: false,
      b: true,
    };
    const result = items.filter(filterOnSingleChoice(map));
    expect(result).toMatchSnapshot();
  });
});

describe('filter on text input', () => {
  const items = [
    { id: '1', text: 'abcdef' },
    { id: '2', text: 'abcfff' },
    { id: '3', text: 'bcdef' },
  ];

  it('filterOnTextInput returns correct for text: abc', () => {
    const map = {
      text: 'abc',
    };
    const result = items.filter(filterOnTextInput(map));
    expect(result).toMatchSnapshot();
  });

  it('filterOnTextInput returns correct for text: abcd', () => {
    const map = {
      text: 'abcd',
    };
    const result = items.filter(filterOnTextInput(map));
    expect(result).toMatchSnapshot();
  });

  it('filterOnTextInput returns correct for text: b', () => {
    const map = {
      text: 'b',
    };
    const result = items.filter(filterOnTextInput(map));
    expect(result).toMatchSnapshot();
  });

  it('filterOnTextInput returns correct for text: e', () => {
    const map = {
      text: 'e',
    };
    const result = items.filter(filterOnTextInput(map));
    expect(result).toMatchSnapshot();
  });
});

describe('filter on multiple choices', () => {
  const items = [
    { id: '1', text: 'abc' },
    { id: '2', text: 'abc' },
    { id: '3', text: 'def' },
    { id: '4', text: 'ghi' },
  ];

  it('filterOnMultipleChoices returns correct for text: abc', () => {
    const map = {
      text: ['abc'],
    };
    const result = items.filter(filterOnMultipleChoices(map));
    expect(result).toMatchSnapshot();
  });

  it('filterOnMultipleChoices returns correct for text: abc, def', () => {
    const map = {
      text: ['abc', 'def'],
    };
    const result = items.filter(filterOnMultipleChoices(map));
    expect(result).toMatchSnapshot();
  });

  it('filterOnMultipleChoices returns correct for text: empty', () => {
    const map = {
      text: [],
    };
    const result = items.filter(filterOnMultipleChoices(map));
    expect(result).toMatchSnapshot();
  });
});
