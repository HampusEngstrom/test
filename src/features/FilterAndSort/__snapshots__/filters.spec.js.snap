// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter on multiple choices filterOnMultipleChoices returns correct for text: abc 1`] = `
Array [
  Object {
    "id": "1",
    "text": "abc",
  },
  Object {
    "id": "2",
    "text": "abc",
  },
]
`;

exports[`filter on multiple choices filterOnMultipleChoices returns correct for text: abc, def 1`] = `
Array [
  Object {
    "id": "1",
    "text": "abc",
  },
  Object {
    "id": "2",
    "text": "abc",
  },
  Object {
    "id": "3",
    "text": "def",
  },
]
`;

exports[`filter on multiple choices filterOnMultipleChoices returns correct for text: empty 1`] = `
Array [
  Object {
    "id": "1",
    "text": "abc",
  },
  Object {
    "id": "2",
    "text": "abc",
  },
  Object {
    "id": "3",
    "text": "def",
  },
  Object {
    "id": "4",
    "text": "ghi",
  },
]
`;

exports[`filter on single choice filterOnSingleChoice returns correct a 1`] = `
Array [
  Object {
    "a": "hej",
    "b": "hej",
    "id": "1",
  },
  Object {
    "a": "hej",
    "b": null,
    "id": "2",
  },
  Object {
    "a": "hej",
    "b": "hej",
    "id": "4",
  },
  Object {
    "a": "hej",
    "b": null,
    "id": "5",
  },
]
`;

exports[`filter on single choice filterOnSingleChoice returns correct a b 1`] = `
Array [
  Object {
    "a": "hej",
    "b": "hej",
    "id": "1",
  },
  Object {
    "a": "hej",
    "b": "hej",
    "id": "4",
  },
]
`;

exports[`filter on single choice filterOnSingleChoice returns correct b 1`] = `
Array [
  Object {
    "a": "hej",
    "b": "hej",
    "id": "1",
  },
  Object {
    "a": null,
    "b": "hej",
    "id": "3",
  },
  Object {
    "a": "hej",
    "b": "hej",
    "id": "4",
  },
  Object {
    "a": null,
    "b": "hej",
    "id": "6",
  },
]
`;

exports[`filter on text input filterOnTextInput returns correct for text: abc 1`] = `
Array [
  Object {
    "id": "1",
    "text": "abcdef",
  },
  Object {
    "id": "2",
    "text": "abcfff",
  },
]
`;

exports[`filter on text input filterOnTextInput returns correct for text: abcd 1`] = `
Array [
  Object {
    "id": "1",
    "text": "abcdef",
  },
]
`;

exports[`filter on text input filterOnTextInput returns correct for text: b 1`] = `
Array [
  Object {
    "id": "3",
    "text": "bcdef",
  },
]
`;

exports[`filter on text input filterOnTextInput returns correct for text: e 1`] = `Array []`;
